ambry-router/src/main/java/com/github/ambry/router/GetBlobOperation.java

```diff
@@ -201,6 +280,15 @@ void abort(Exception abortCause) {
         blobDataChannel.completeRead();
       }
     }
+    switch(((RouterException) abortCause).getErrorCode()) {
+      /**
+       * If the blob is not found, then delete its metadata from frontend cache.
```

Comment:
nit: probably use end of line comments ( // If the blob is not found ... ) or block comments (/* * */) for multiple lines since we seem to do that in rest of code?

Reply:
agreed, will fix